vae.py

```diff
@@ -276,7 +276,12 @@ def compute_loss(self, latent_mean, latent_logvar, vae_prev_obs, vae_next_obs, v
         """
 
         num_unique_trajectory_lens = len(np.unique(trajectory_lens))
-        assert (num_unique_trajectory_lens == 1) or (self.args.vae_subsample_elbos and self.args.vae_subsample_decodes)
+
+        if 'Humanoid' in self.args.env_name:
+            # The agent can die in the humanoid environment, so traj. lengths will not be the same but that's alright.
```

Comment:
It's not alright to call this function in that case: what it will do is cut off all trajectories at the *max* trajectory length, which means that many trajectories will have zeros (as they're zero-padded). This function can only be called when we can safely batch all trajectories, either because the trajectories are all the same length, or because we're subsampling a fixed number of elbos and decodes. If these conditions cannot be met, the flag "split_batches_by_elbo" has to be set to True.

Reply:
Ah I see -- I think I misunderstood the code in that case. I'll look into this again and set the parameters and rerun the experiments.

Reply:
Thanks! I'll add a comment to the assertion to explain why it's there :)